### PR TITLE
move a bunch of chat code into its own package

### PR DESCRIPTION
**Pull request by mmaxim**
@patrickxb @songgao 

The point of this is to move a lot of chat support code into it's own package. This makes service less cluttered with chat code, and gives a natural place to stick new stuff (like chat_storage.go).

Let me know what you think.

### CODE REVIEW

File: go/chat/chat_boxer.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2016 Keybase, Inc. All rights reserved. Use of
 // this source code is governed by the included BSD license.
 
-package service
+package chat
```
Comment (patrickxb) [inline go/chat/chat_boxer.go]: rename the file to just boxer.go

Comment (mmaxim) [inline go/chat/chat_boxer.go]: Yeah good call.

File: go/chat/mocks.go [hunk 2]
```diff
@@ -1,7 +1,4 @@
-// Copyright 2016 Keybase, Inc. All rights reserved. Use of
-// this source code is governed by the included BSD license.
-
-package service
+package chat
```
Comment (songgao) [inline go/chat/mocks.go]: Maybe change this file name to `mocks_test.go` so it only builds with test?

**Comment by mmaxim** [comment]
We can handle chat_local.go later, but some of that could be busted up as well. Thanks!

**Pull request merged**